modules/ui/feature_list.js

```diff
@@ -122,14 +122,16 @@ export function uiFeatureList(context) {
 
             if (!q) return result;
 
-            var idMatch = q.match(/^([nwr])([0-9]+)$/);
+            var idMatch = q.match(/^(.*[^\w])?(node|n|way|w|relation|r)[^\w]?([0-9]+)([^\w].*)?$/i);
 
             if (idMatch) {
+                var elemType = idMatch[2][0];
```

Comment:
I assume this is intending to take the first character of the string (index 0)?

If so, this is invalid syntax in Javascript (I think...it's been a while), I'd use `.charAt(0)`.

Reply:
I'm not experience in JS, thanks for pointing this out. I'll commit a change (including this) with the result of your other feedback.